### PR TITLE
Added unit test for InsecureJSFViewStatePassiveScanner 

### PR DESCRIPTION
**Pull request by rscipien**
Test for issue [2807](https://github.com/zaproxy/zaproxy/issues/2807) + fix small bug checking value of view state instead of id

### CODE REVIEW

File: test/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScannerUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+public class InsecureJSFViewStatePassiveScannerUnitTest extends PassiveScannerTest {
+
+    @Override
+    protected PluginPassiveScanner createScanner() {
+        return new InsecureJSFViewStatePassiveScanner();
+    }
+    
+    
+    @Test
+    public void noResponsBody() throws HttpMalformedHeaderException {
```
Comment (kingthorin) [inline test/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScannerUnitTest.java]: Response is missing an e

File: test/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScannerUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+         msg.setResponseBody("");
+         setTextHtmlResponsHeader(msg);
+         rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+         assertThat(alertsRaised.size(), equalTo(0));    
+    }
+    
+    
+    @Test
+    public void ignonrUnderscoreValues() throws HttpMalformedHeaderException {
```
Comment (kingthorin) [inline test/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScannerUnitTest.java]: There's a typo in ignore

File: test/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScannerUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                + "<input type='hidden' id='javax.faces.viewstate' value='" + encoded + "'/>"
+                + "</body>"
+                + "</html>");
+        setTextHtmlResponsHeader(msg);
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(1));    
+    }
+
+    private void setTextHtmlResponsHeader(HttpMessage msg) throws HttpMalformedHeaderException {
```
Comment (kingthorin) [inline test/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScannerUnitTest.java]: Response is missing an e here too

**Comment by rscipien** [comment]
How to check which test failed on travis? On my machine all test are ok

File: test/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScannerUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Content-Type: text/html;charset=UTF-8\r\n" +
+                "Content-Length: 0\r\n");
+    }
+    
+    private byte[] deflateViewState(String value) throws IOException {
```
Comment (kingthorin) [inline test/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScannerUnitTest.java]: The input param `value` doesn't seem to be used

**Comment by kingthorin** [comment]
Tests failing

**Comment by kingthorin** [comment]
Please do an interactive rebase and fix-up or Squash into a single "atomic" commit: https://github.com/zaproxy/zaproxy/blob/develop/CONTRIBUTING.md#what-we-zap-team-expect-from-you .

- https://www.kernel.org/pub/software/scm/git/docs/git-rebase.html#_interactive_mode
- https://wiki.eclipse.org/EGit/User_Guide#Interactive_RebaseLeave a comment

**Comment by rscipien** [comment]
Fixed test failing. Added checking if state is stored on server. Small change checking for state view id.

**Comment by rscipien** [comment]
Match unit test guidelines https://github.com/zaproxy/zaproxy/wiki/DevGuidelines#unit-test-specific-guidelines

File: src/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScanner.java [hunk 5]
```diff
@@ -172,28 +173,6 @@ private boolean isViewStateSecure(String viewState, String charset) {
...
-			// /////////////////////////////
-			// Step 3
-			// Try to determine if ViewState is encrypted or contains clear text
-			// strings.
-
-
-			return isRawViewStateSecure(out.toString());
-		} catch (IOException ex) {
-		}
-
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScanner.java]: Why was this block removed?

**Comment by rscipien** [comment]
The same functionality is in [Base64.decode(String,int)](https://github.com/rscipien/zap-extensions/blob/d3375c31ba9cb488dd98a6e3058ca44b21cb217a/src/org/zaproxy/zap/extension/pscanrulesBeta/InsecureJSFViewStatePassiveScanner.java#L161)

**Comment by kingthorin** [comment]
TODO:

- [ ] Update manifest to include scanner changes

**Comment by kingthorin** [comment]
Don't merge before TODO is done

**Comment by kingthorin** [comment]
Will conflict after https://github.com/zaproxy/zap-extensions/pull/539 is accepted. I can tweak the manifest when that happens.

**Comment by kingthorin** [comment]
Conflict fixed.

@psiinon, leaving to you for merge since I ended up working on it.

**Comment by thc202** [comment]
@rscipien, @kingthorin thanks!

**Pull request merged**